src/navx_node.cpp

```diff
@@ -2,25 +2,50 @@
  * navx_node.cpp
  * Runs the Kauai Labs NavX, using modified NavX library
  * VERSION: 1.0.0
- * Last changed: 2019-10-05
+ * Last changed: 2019-10-31
  * Authors: Jude Sauve <sauve031@umn.edu>
  * Maintainers: Nick Schatz <schat127@umn.edu>
  * MIT License
  * Copyright (c) 2019 UMN Robotics
  */
 
+/* Interface: 
+ *  Pub: 
+ *   imu/data (sensor_msgs/Imu)
+ *   imu/euler (geometry_msgs/Point)
+ * Param: 
+ *   frequency (double) 50.0; The frequency of the read loop
+ *   publish_euler (bool) false; Whether to publish euler orientation
+ *   device_path (string) /dev/ttyACM0; The device serial port path
+ *   frame_id (string) imu_link; The Imu message header frame ID
+ *   covar_samples (int) 100; The number of samples to store to calculate covariance
+ */
+
 // ROS Libs
 #include <ros/ros.h>
 #include <sensor_msgs/Imu.h>
 #include <geometry_msgs/Point.h>
+#include "boost/array.hpp"
 
 // Native_Libs
 #include <string>
+#include <cmath>
 
 // Custom_Libs
 #include "ahrs/AHRS.h"
 
-#define DEG_TO_RAD ((2.0 * 3.14159) / 360.0)
+static const float DEG_TO_RAD = M_PI / 180.0F;
+static const float GRAVITY = 9.81F; // m/s^2, if we actually go to the moon remember to change this
+typedef struct {
+  float ypr[3];
+  float ang_vel[3];
+  float accel[3];
+} OrientationEntry;
+
+// Function Prototypes
+bool calculate_covariance(boost::array<double, 9> &orientation_mat, 
+                          boost::array<double, 9> &ang_vel_mat, 
+                          boost::array<double, 9> &accel_mat);
 
```

Comment:
As you can see, this should all now go under Global Variables. If there's a lot of function prototypes, then I would add another comment and break them out, else leave them as a "variable". 
Custom types should go after Params & before Global Vars, b/c you can't have a param that's more complex than a basic type*, same with the other ros stuff defined before. 
The point is to keep the boilerplate in as consistent and priority-sorted order; even the functions after main() should go in (roughly) the order of complexity, because the smallest of helper functions will never need be changed nearly as much. 
(* other than vectors or maps, or a custom advanced ros type, but that's for config files, and never needed unless you are doing very unusual stuff; as for me, it's usually been for testing purposes, not normal).